src/main/java/seedu/guest/commons/util/StringUtil.java

```diff
@@ -59,15 +59,52 @@ public static boolean containsDateIgnoreHyphenIgnoreSpace(String sentence, Strin
         if (preppedDate.isEmpty()) {
             return false;
         }
-        checkArgument(preppedDate.split("\\s+").length == 1, "date parameter should be a single date");
+        checkArgument(preppedDate.split("\\s+").length == 1, "Date parameter should be a single date");
 
         String preppedSentence = sentence.replace("-", "");
-        String[] wordsInPreppedSentence = preppedSentence.split("\\s+");
+        String[] datesInPreppedSentence = preppedSentence.split("\\s+");
 
-        return Arrays.stream(wordsInPreppedSentence)
+        return Arrays.stream(datesInPreppedSentence)
                 .anyMatch(preppedDate::equalsIgnoreCase);
     }
 
+    /**
+     * Returns true if the {@code sentence} contains the {@code isRoomClean}.
+     *   Ignores case, but must be either 'yes', 'no', 'y' or 'n'.
+     *   <br>examples:<pre>
+     *       containsIsRoomCleanIgnoreCase("yes", "y") == true
+     *       containsIsRoomCleanIgnoreCase("yes", "yes") == true
+     *       containsIsRoomCleanIgnoreCase("yes", "yEs") == true
+     *       containsIsRoomCleanIgnoreCase("no", "n") == true
+     *       containsIsRoomCleanIgnoreCase("no", "no") == true
+     *       containsIsRoomCleanIgnoreCase("no", "nO") == true
+     *       containsIsRoomCleanIgnoreCase("yes", "yep") == false //not in the correct format
+     *       containsIsRoomCleanIgnoreCase("no", "nope") == false //not in the correct format
+     *       </pre>
+     * @param sentence cannot be null
+     * @param isRoomClean cannot be null, cannot be empty, must be a single isRoomClean
+     */
+    public static boolean containsIsRoomCleanIgnoreCase(String sentence, String isRoomClean) {
+        requireNonNull(sentence);
+        requireNonNull(isRoomClean);
+
+        String preppedIsRoomClean = isRoomClean.trim();
+        checkArgument(!preppedIsRoomClean.isEmpty(), "IsRoomClean parameter cannot be empty");
+        checkArgument(preppedIsRoomClean.split("\\s+").length == 1,
+                "IsRoomClean parameter should be a single IsRoomClean");
+        if (preppedIsRoomClean.equals("y") || preppedIsRoomClean.equals("Y")) {
```

Comment:
For "y", "n", "yes", "no", should we abstract them out into variables to prevent floating literals? for example, "y" could be abbreviatedRoomClean and "yes" could be standardisedRoomClean?

Reply:
Agreed.